freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/poller/UserSyncPollerEntitlementChecker.java

```diff
@@ -0,0 +1,23 @@
+package com.sequenceiq.freeipa.service.freeipa.user.poller;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+
+@Component
+class UserSyncPollerEntitlementChecker {
+    @VisibleForTesting
+    static final String INTERNAL_ACTOR_CRN = new InternalCrnBuilder(Crn.Service.IAM).getInternalCrnForServiceAsString();
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    public boolean isAccountEntitled(String accountId) {
```

Comment:
just an idea: we could add `@Cacheable` here so we won't dos UMS

Reply:
The underlying GrpcUmsClient.getAccountDetails method is cached in the umsAccountCache